### PR TITLE
DeviceStore

### PR DESCRIPTION
**Pull request by jpnurmi**
DeviceStore replaces the former monolithic FirmwareModel. It provides a list of devices and that's it. FwupdNotifier is now dedicated to the top-level fwupd properties. We don't need device update monitoring on this level anymore because it's provided by DeviceModel.

### CODE REVIEW

File: test/device_store_test.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      flags: const {FwupdDeviceFlag.updatable},
+    );
+    final dev2 = FwupdDevice(
+      deviceId: '2',
+      name: '',
+      plugin: '',
+      flags: const {FwupdDeviceFlag.updatableHidden},
+    );
+    final dev3 = FwupdDevice(
+      deviceId: '2',
```
Comment (d-loose) [inline test/device_store_test.dart:37]: Completely irrelevant, but `dev3` should probably have `deviceId: '3'`

**Review by d-loose — APPROVED** [review]
Simplifies everything a lot :+1:

File: test/device_store_test.dart (hunk 1 above)
Comment (jpnurmi) [reply test/device_store_test.dart]: would've been fun if the test started failing after fixing the id

**Comment by jpnurmi** [comment]
I hope I didn't create massive conflicts with these disruptive changes lately...

**Pull request merged**